### PR TITLE
Kirkstone migration and updates

### PR DESCRIPTION
**Pull request by HaberkornJonas**
### Changes Overview

- **[Migrating to kirkstone and made build pass.](https://github.com/axians/meta-microservicebus/commit/b947e71d155de11413ba13981de8df91b8135964)** (b947e71d155de11413ba13981de8df91b8135964)
  - Adapted the syntax (released in [Yocto 3.4](https://docs.yoctoproject.org/dev/migration-guides/release-3.4.html))
  - Edited LAYERSERIES_COMPAT to be kirkstone
  - Fixed QA_EMPTY_DIRS issue as /run wasn't empty due to msb-dam
  - DIRFILES disowning parent dir of files added/modified by msb-node-user
  - Recreated serialport recipe with devtool and updated to v10.4.0
  - Recreated dbus recipe with devtool

- **[Merging newer changes from default warrior branch into kirkstone.](https://github.com/axians/meta-microservicebus/commit/d640e2e7952e83e36b36490af165be817a86028b)** (d640e2e7952e83e36b36490af165be817a86028b)
  - Updated user password handling (see: https://github.com/axians/meta-microservicebus/commit/2ee85c24ca325f3096167b4dbc0acd009f6cd5fa)
  - Added support for NODE_EXTRA_CA_CERT (see: https://github.com/axians/meta-microservicebus/commit/042778bdb50236f2d8420341d2144b2838ade0fd)
  - Migrated msb-node-user 2.1.0 to support kirkstone
  - Resolved warning "microservicebus-node-service: recipe doesn't inherit features_check"
  - Resolved warning "the license listed Unknown was not in the licenses collected for recipe msb-node"

- **[Added recipe for msb-node@3.4.5](https://github.com/axians/meta-microservicebus/commit/1149c10556eb3d036e826418bf883b0d95d6ed3f)** (1149c10556eb3d036e826418bf883b0d95d6ed3f)

- **[Added inherit features_check to fix warning during build of npm-dam-scheduled-service](https://github.com/axians/meta-microservicebus/commit/57efc48b5c52ec4a918f321dc5f558bceeac3f80)** (57efc48b5c52ec4a918f321dc5f558bceeac3f80)

- **[Added recipe for nodejs v14 as v16 provided by meta-oe isn't supported by msb-node yet](https://github.com/axians/meta-microservicebus/commit/ddc54533a1a8439a430cb5c1a04f1ab8a3936047) (ddc54533a1a8439a430cb5c1a04f1ab8a3936047)**
  - Can be removed once this is fixed: https://github.com/axians/microservicebus-core/issues/91

### CODE REVIEW

File: meta-microservicebus-node/recipes-microservicebus-node/nodejs/nodejs_14.18.1.bb [hunk 1]
```diff
@@ -0,0 +1,200 @@
+DESCRIPTION = "nodeJS Evented I/O for V8 JavaScript"
```
Comment (jonaskgandersson) [inline meta-microservicebus-node/recipes-microservicebus-node/nodejs/nodejs_14.18.1.bb:1]: Better to use nodejs 16.14.2 as is delivered in Kirkstone and solve the issue in msb-core instead.
msb-core issue [91](https://github.com/axians/microservicebus-core/issues/91)

File: meta-microservicebus-node/recipes-microservicebus-node/microservicebus-node-service/microservicebus-node-service_2.4.0.bb [hunk 2]
```diff
@@ -23,16 +23,18 @@ MSB_NODE_USER ?= "msb"
...
-             
+
     #Replace parameters in service file
     sed -i -e 's:@MSB_NODE_ARG@:${MSB_NODE_ARG}:g' ${WORKDIR}/microservicebus-node.service
     sed -i -e 's:@MSB_NODE_WORK_DIR@:${MSB_NODE_WORK_DIR}:g' ${WORKDIR}/microservicebus-node.service
     sed -i -e 's:@MSB_NODE_USER@:${MSB_NODE_USER}:g' ${WORKDIR}/microservicebus-node.service
     sed -i -e 's:@MSB_NODE_GROUP@:${MSB_NODE_GROUP}:g' ${WORKDIR}/microservicebus-node.service
     sed -i -e 's:@MSB_NODE_HOST@:${MSB_NODE_HOST}:g' ${WORKDIR}/microservicebus-node.service
     sed -i -e 's:@MSB_DAM_SOCKETPATH@:${MSB_DAM_SOCKETPATH}:g' ${WORKDIR}/microservicebus-node.service
+    sed -i -e 's:@MSB_NODE_EXTRA_CA_CERTS@:${MSB_NODE_EXTRA_CA_CERTS}:g' ${WORKDIR}/microservicebus-node.service
```
Comment (wmmihaa) [inline meta-microservicebus-node/recipes-microservicebus-node/microservicebus-node-service/microservicebus-node-service_2.4.0.bb:37]: Can you describe how this is used in local.conf

File: meta-microservicebus-node/recipes-microservicebus-node/microservicebus-node-service/microservicebus-node-service_2.4.0.bb [hunk 3]
```diff
@@ -23,16 +23,18 @@ MSB_NODE_USER ?= "msb"
 MSB_NODE_GROUP ?= "msb"
 MSB_NODE_HOST ?= "microservicebus.com"
 MSB_DAM_SOCKETPATH ?= "/var/run/dam"
+MSB_NODE_EXTRA_CA_CERTS ?= ""
```
Comment (jonaskgandersson) [inline meta-microservicebus-node/recipes-microservicebus-node/microservicebus-node-service/microservicebus-node-service_2.4.0.bb:26]: In local.conf set MSB_NODE_EXTRA_CA_CERTS = "PATH-TO-CA.crt"

File: meta-microservicebus-node/recipes-microservicebus-node/node-pty/node-pty_0.9.0.bb [hunk 4]
```diff
@@ -33,7 +33,7 @@ inherit npm
```
Comment (jonaskgandersson) [inline meta-microservicebus-node/recipes-microservicebus-node/node-pty/node-pty_0.9.0.bb]: Remove node-pty recipe, msb-core fetch this on demand.

Comment (jonaskgandersson) [inline meta-microservicebus-node/recipes-microservicebus-node/node-pty/node-pty_0.9.0.bb]: Remove node-pty recipe, msb-core fetch this on demand.

File: meta-microservicebus-node/recipes-microservicebus-node/nodejs-dbus/nodejs-dbus_0.4.0.bb [hunk 5]
```diff
@@ -1,78 +1,80 @@
+# Recipe created by recipetool
```
Comment (jonaskgandersson) [inline meta-microservicebus-node/recipes-microservicebus-node/nodejs-dbus/nodejs-dbus_0.4.0.bb]: Remove nodejs-dbus recipe, msb-core fetch this on demand.

Comment (jonaskgandersson) [inline meta-microservicebus-node/recipes-microservicebus-node/nodejs-dbus/nodejs-dbus_0.4.0.bb]: Remove nodejs-dbus recipe, msb-core fetch this on demand.

File: meta-microservicebus-node/recipes-microservicebus-node/nodejs-serialport/nodejs-serialport_10.4.0.bb [hunk 6]
```diff
@@ -0,0 +1,75 @@
+# Recipe created by recipetool
```
Comment (jonaskgandersson) [inline meta-microservicebus-node/recipes-microservicebus-node/nodejs-serialport/nodejs-serialport_10.4.0.bb]: Remove node-serialport recipe, msb-core fetch this on demand.

Comment (jonaskgandersson) [inline meta-microservicebus-node/recipes-microservicebus-node/nodejs-serialport/nodejs-serialport_10.4.0.bb]: Remove node-serialport recipe, msb-core fetch this on demand.

File: meta-microservicebus-node/conf/layer.conf [hunk 7]
```diff
@@ -11,9 +11,12 @@ BBFILE_PRIORITY_meta-microservicebus-node= "16"
...
-LAYERSERIES_COMPAT_meta-microservicebus-node = "warrior dunfell"
+LAYERSERIES_COMPAT_meta-microservicebus-node = "kirkstone"
 
 # Additional license directories.
 LICENSE_PATH += "${LAYERDIR}/files/common-licenses"
 SPDXLICENSEMAP[Unlicensev1] = "Unlicense"
 SPDXLICENSEMAP[WTFPLv1] = "WTFPL"
+
+# Using NodeJS v14 until v16+ is supported
+PREFERRED_VERSION:nodejs ?= "14.18.1"
```
Comment (jonaskgandersson) [inline meta-microservicebus-node/conf/layer.conf]: @wmmihaa is fixing nodejs v16 support.

Comment (jonaskgandersson) [inline meta-microservicebus-node/conf/layer.conf]: @wmmihaa is fixing nodejs v16 support.

File: meta-microservicebus-node/recipes-microservicebus-node/microservicebus-node/microservicebus-node_3.4.3.bb [hunk 8]
```diff
@@ -259,107 +259,107 @@ S = "${WORKDIR}/npm"
```
Comment (jonaskgandersson) [inline meta-microservicebus-node/recipes-microservicebus-node/microservicebus-node/microservicebus-node_3.4.3.bb]: Will get new msb-node version for nodejs V16

Comment (jonaskgandersson) [inline meta-microservicebus-node/recipes-microservicebus-node/microservicebus-node/microservicebus-node_3.4.3.bb]: Will get new msb-node version for nodejs V16

**Review by jonaskgandersson — COMMENTED** [review]
We will add support for node v16 instead of changing version in Yocto.

We can remove recipes:

- node-pty
- nodejs-dbus
- nodejs-serialport

As we never install them in image, msb-core installs them on demand.

File: meta-microservicebus-node/recipes-microservicebus-node/nodejs-serialport/nodejs-serialport_10.4.0.bb (hunk 6 above)
Comment (HaberkornJonas) [reply meta-microservicebus-node/recipes-microservicebus-node/nodejs-serialport/nodejs-serialport_10.4.0.bb]: * Removed the **node-serialport** recipe in https://github.com/axians/meta-microservicebus/pull/49/commits/ec66b71e75e941b89f45fdb6c8329761356469eb
* Removed **packagegroup-microservicebus-extra** in https://github.com/axians/meta-microservicebus/pull/49/commits/7cbc4c41d07079808d5eedc865e355852d813e7c as its only dependency (node-serialport) was removed
* Edited **README** to remove occurrences of node-serialport and microservicebus-extra

Comment (HaberkornJonas) [reply meta-microservicebus-node/recipes-microservicebus-node/nodejs-serialport/nodejs-serialport_10.4.0.bb]: * Removed the **node-serialport** recipe in https://github.com/axians/meta-microservicebus/pull/49/commits/ec66b71e75e941b89f45fdb6c8329761356469eb
* Removed **packagegroup-microservicebus-extra** in https://github.com/axians/meta-microservicebus/pull/49/commits/7cbc4c41d07079808d5eedc865e355852d813e7c as its only dependency (node-serialport) was removed
* Edited **README** to remove occurrences of node-serialport and microservicebus-extra

File: meta-microservicebus-node/recipes-microservicebus-node/nodejs-dbus/nodejs-dbus_0.4.0.bb (hunk 5 above)
Comment (HaberkornJonas) [reply meta-microservicebus-node/recipes-microservicebus-node/nodejs-dbus/nodejs-dbus_0.4.0.bb]: * Removed the **nodejs-dbus** recipe in https://github.com/axians/meta-microservicebus/pull/49/commits/fdad0fea0aeab1970fff0462c9923a79b2d68b34
* Edited **README** to remove occurrences of nodejs-dbus

File: meta-microservicebus-node/recipes-microservicebus-node/node-pty/node-pty_0.9.0.bb (hunk 4 above)
Comment (HaberkornJonas) [reply meta-microservicebus-node/recipes-microservicebus-node/node-pty/node-pty_0.9.0.bb]: Removed the **nodejs-pty** recipe in https://github.com/axians/meta-microservicebus/pull/49/commits/16ef03fa627f06aa6b8dd4353f0702cdbc87acd5

File: meta-microservicebus-node/recipes-microservicebus-node/nodejs/nodejs_14.18.1.bb (hunk 1 above)
Comment (HaberkornJonas) [reply meta-microservicebus-node/recipes-microservicebus-node/nodejs/nodejs_14.18.1.bb:1]: I agree.

Should I remove this nodejs v14 recipe and `PREFERRED_VERSION:nodejs ?= "14.18.1"` from layer.conf or do you want to keep it until https://github.com/axians/microservicebus-core/issues/91 is implemented?

I added it for convenience but can also move it to our app layer if you prefer to not have it here.

File: meta-microservicebus-node/recipes-microservicebus-node/microservicebus-node-service/microservicebus-node-service_2.4.0.bb (hunk 2 above)
Comment (HaberkornJonas) [reply meta-microservicebus-node/recipes-microservicebus-node/microservicebus-node-service/microservicebus-node-service_2.4.0.bb:37]: We use it like this in local.conf: 
`MSB_NODE_EXTRA_CA_CERTS = "/usr/local/share/ca-certificates/zscaler-root-ca.crt"`

It's currently the certificate from the [meta-vinci-energies](https://github.com/axians/meta-vinci-energies) layer but we might need to add other certificates at some point and would change the name of the file in that case.

**Comment by HaberkornJonas** [comment]
@wmmihaa, @jonaskgandersson - I saw that you merged #48 into the warrior branch, as it's the default branch I also merged it here to be up-to-date with the latest changes (8e54b2bc0c08d547cf1154002cea566d865e4a47) and fixed a features_check warning (84958f1729f6833d081f245262fc037da8a53fa3)

**Pull request merged**